impl/sql/appender/trillian.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appender
+
+import (
+	"crypto"
+	"encoding/json"
+
+	"github.com/google/keytransparency/core/appender"
+	"github.com/google/keytransparency/core/transaction"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/merkle/rfc6962"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// Trillian is an appender to a Trillian Log.
+type Trillian struct {
+	logID int64
+	log   *client.LogClient
+	send  bool
+}
+
+// NewTrillian creates a new client to a Trillian Log.
+// hc is the underlying HTTP client use to communicated with
+// CT. If hc is nil, CT will create a default HTTP client.
+func NewTrillian(ctx context.Context, logID int64, cc *grpc.ClientConn, pubkey crypto.PublicKey) (appender.Appender, error) {
+	send := cc != nil
+	var logClient *client.LogClient
+	if send {
+		cli := trillian.NewTrillianLogClient(cc)
+		hasher := rfc6962.TreeHasher{Hash: crypto.SHA256}
+		logClient = client.New(logID, cli, hasher, pubkey)
+	}
+
+	return &Trillian{
+		logID: logID,
+		log:   logClient,
+		send:  send,
+	}, nil
+}
+
+// Append sends obj to Trillian as a json object at the given epoch index.
+func (t *Trillian) Append(ctx context.Context, txn transaction.Txn, epoch int64, obj interface{}) error {
+	if t.send {
+		b, err := json.Marshal(obj)
+		if err != nil {
+			return err
+		}
+		// TODO(gbelvin): Add leaf at a specific index. trillian#423
+		if err := t.log.AddLeaf(ctx, b); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Epoch retrieves a given object at a particular index.
+func (t *Trillian) Epoch(ctx context.Context, epoch int64, obj interface{}) ([]byte, error) {
```

Comment:
I'm not sure but I think tests will not pass without implementing `Epoch` and `Latest`, right?